src/tests/ftest/aggregation/aggregation_throttling.py

```diff
@@ -0,0 +1,151 @@
+        """
+
+        # Create pool and container
+        self.update_ior_cmd_with_pool()
+
+        # Disable the aggregation
+        self.pool.set_property("reclaim", "disabled")
+
+        # Run ior with -k option to retain the file and not delete it
+        out = self.run_ior_with_pool()
+        metric_before_aggregate = IorCommand.get_ior_metrics(out)
+
+        # Run ior the second time on the same pool and container, so another
+        # copy of the file is inserted in DAOS.
+        out = self.run_ior_with_pool(update=False)
+
+        # wait 90 seconds for files to get old enough for aggregation
+        self.log.info("Waiting for 90 seconds for aggregation to start")
+        time.sleep(90)
+        # Enable the aggregation
+        self.pool.set_property("reclaim", "time")
+
+        # Run ior the third time while the aggregation of first two runs
+        # are running in the background.
+        out = self.run_ior_with_pool(update=False)
+        metric_after_aggregate = IorCommand.get_ior_metrics(out)
+
+        # When DAOS-5057 is fixed, adjust the percentage. For now,
+        # keep it at 30 %
+        expected_perf_diff = 30.0
+
+        self.verify_performance(metric_before_aggregate,
+                                metric_after_aggregate,
+                                0, # write_perf
+                                expected_perf_diff) # 30% perf difference
+
+        self.verify_performance(metric_before_aggregate,
+                                metric_after_aggregate,
+                                1, # read_perf
+                                expected_perf_diff)
+
+    def verify_performance(self, before_metric, after_metric, read_write_idx,
+                           expected_perf_diff):
+        """Verify the after_metric read/write performances are within
+           +/- expected_perf_diff % of before_metric_performance.
+
+        Args:
+           before_metric (IorMetrics): ior metrics before aggregation
+           after_metric (IorMetrics): ior metrics in concurrent with aggregation
+           read_write_idx (int): Read (1) or Write (0) index in the IorMetrics
+           expected_perf_diff (float): Expected performance difference between
+                                       before_metric and after_metric.
+        """
+
+        self.log.info("\n\n%s Performance" %
+                      "Read" if read_write_idx else "Write")
+
+        max_mib = int(IorMetrics.Max_MiB)
+        min_mib = int(IorMetrics.Min_MiB)
+        mean_mib = int(IorMetrics.Mean_MiB)
+
+        max_prev = float(before_metric[read_write_idx][max_mib])
+        max_curr = float(after_metric[read_write_idx][max_mib])
+        self.log.info("max_prev = %f, max_curr = %f" % (
+            max_prev, max_curr))
+
+        min_prev = float(before_metric[read_write_idx][min_mib])
+        min_curr = float(after_metric[read_write_idx][min_mib])
+        self.log.info("min_prev = %f, min_curr = %f" % (
```

Comment:
(pylint-logging-not-lazy) Specify string format arguments as logging function parameters